### PR TITLE
feat(*)!: Unifies all calls behind the ctl interface topics

### PR DESCRIPTION
**Pull request by thomastaylor312**
## Feature or Problem

This essentially backs out all of the changes I added to 0.30 of the control interface. There are two main reasons behind this:

First, and most important, a NATS KV bucket and a NATS topic are two different security contexts. In multitenant environments (like Cosmonic, which is where we ran into this), it is perfectly reasonable for a platform operator to have access to the "API" of the thing they are running (the ctl interface in this case), but not access to the underlying datastore. In order to give access to buckets, you'd have to explicitly grant permissions and resign tokens for each new lattice or NATS account (depending on your security boundary) that you spin up.

Second, this unifies everything behind a single API - the NATS topics. Although NATS KV is still NATS, it is a weird API to essentially switch from requests to a "please connect to the database." It also means that _all_ clients have to know about how we encode and store the data, as well as other implementation details. Seeing as that might change in the future (with things like linkdefs 2.0), it seems better to have this behind an API.

You might be saying at this point, "but wait Taylor! Didn't we do this to fix some issues?" Yes, we did. However, we are actually ending up with the same behavior as currently exists in the client. The Rust host just proxies through the linkdef put information and puts the key in the database. It doesn't update its local cache until it receives the value update from the bucket. It also returns its cached claims on a get links command, which is exactly what happens with the caching as it worked before in the client. So we should be in the exact state we were before, but with a cleaner API.

Also, while I was here, I cleaned up some more of the code to be more concise and removed behavior from starting a provider that tried to automatically auction on a host. None of the other functions attempted to do that, nor should they IMO. That behavior is nice to have in something like wash, but an API client represents lower level API interactions. So I removed that from the `start_provider` function.

## Release Information

0.31

## Consumer Impact

BREAKING CHANGE: Client and ClientBuilder no longer require a KV store
BREAKING CHANGE: Client::new and ClientBuilder::build are no longer fallable
BREAKING CHANGE: `start_provider` no longer attempts to auction a host

## Testing

I tested this by updating Wadm to use it and running the e2e tests locally. Since wadm pretty much exercises the full API, this should work. Also, most of the code was copied from the previously working code that just used the topics

### CODE REVIEW

**Comment by thomastaylor312** [comment]
Please note that I'll follow this up with a chore PR to bump the version

File: src/lib.rs [hunk 1]
```diff
@@ -4,42 +4,36 @@
...
 
-use crate::kv::KvStore;
 use crate::otel::OtelHeaderInjector;
 
 type Result<T> = ::std::result::Result<T, Box<dyn std::error::Error + Send + Sync>>;
 
 /// Lattice control interface client
 #[derive(Clone)]
-pub struct Client<T: Clone> {
+pub struct Client {
```
Comment (connorsmith256) [inline src/lib.rs:28]: This is a generic comment.

File: src/lib.rs [hunk 2]
```diff
@@ -243,18 +165,23 @@ impl<T: KvStore + Clone + Send + Sync> Client<T> {
             broker::queries::host_inventory(&self.topic_prefix, &self.lattice_prefix, host_id);
         debug!("get_host_inventory:request {}", &subject);
         match self.request_timeout(subject, vec![], self.timeout).await {
-            Ok(msg) => {
-                let hi: HostInventory = json_deserialize(&msg.payload)?;
```
Comment (connorsmith256) [inline src/lib.rs:247]: bye

File: src/types.rs [hunk 3]
```diff
@@ -77,6 +77,37 @@ pub struct CtlOperationAck {
...
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct GetClaimsResponse {
+    pub claims: Vec<HashMap<String, String>>,
+}
+
+/// A safety type that can handle claims values that are set to Null. This is
+/// only used internally for compatibility, we will still send out [GetClaimsResponse]
+/// as the response type.
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub(crate) struct SafeClaimsResponse {
```
Comment (connorsmith256) [inline src/types.rs:90]: I'm curious if we still need this? I don't think the new host passes null for any field: https://github.com/wasmCloud/wasmCloud/blob/main/crates/host/src/wasmbus/mod.rs#L3904

**Review by connorsmith256 — APPROVED** [review]
LGTM. A couple whimsical comments to cheer you up, and one actual question

File: src/types.rs (hunk 3 above)
Comment (thomastaylor312) [reply src/types.rs:90]: We might not. Will give it a test with wadm and see what happens

**Review by brooksmtownsend — APPROVED** [review]

File: src/types.rs (hunk 3 above)
Comment (thomastaylor312) [reply src/types.rs]: Yep, don't need it. It has been removed

**Pull request merged**